main.yml

```diff
@@ -1,12 +1,22 @@
 - name: Main
-  hosts: '{{ target }}'
+  hosts: "{{ target }}"
   gather_facts: false
   vars_files:
-    - '{{ var_file }}'
+    - "{{ var_file }}"
   roles:
     - node_install
     - node_initialize
     - node_configure
     - node_launch
     - role: support_sync_snapshot
-      when: snapshot_url is defined
\ No newline at end of file
+      when: snapshot_url is defined
+    - role: support_prometheus
+      when: prometheus is defined and prometheus | bool
+- name: Setup Prometheus Monitoring on Chain
+  hosts: telemetry
+  gather_facts: true
```

Comment:
do you need to father facts? You will gather facts if your playbook runs tasks based on remote host values.

This module is automatically called by playbooks to gather useful variables about remote hosts.

Reply:
I think I do not need to gather facts.